### DISCUSSION TITLE
Plugins cannot process OpenAI API key

### DISCUSSION THREAD

**Discussion by sirlaurie** [danny-avila/LibreChat#1646] [Troubleshooting] [answered]
### What happened?

after set OpenAI API key in plugins, start a new conversion, it says
```
error: [handleAbortError] AI response error; aborting request: Unexpected token 's', "sk-NY9Cgvb"... is not valid JSON
```

### Steps to Reproduce

1. set OpenAI API key in plugins
2. submit it
3. start new conversion

### What browsers are you seeing the problem on?

Firefox

### Relevant log output

```shell
2024-01-26 01:58:52 error: [handleAbortError] AI response error; aborting request: Unexpected token 's', "sk-"... is not valid JSON
```

### Screenshots

<img width="952" alt="screenshot 6" src="https://github.com/danny-avila/LibreChat/assets/8928155/b4e4ea6f-ce4e-4ee0-8dae-3e250b707a22">

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
It just worked for me:
![image](https://github.com/danny-avila/LibreChat/assets/110412045/c9ccef34-378e-4176-88fe-792bc8bbd7c1)

![image](https://github.com/danny-avila/LibreChat/assets/110412045/37b60bc9-dc01-4d3f-bf3d-8b082b1d902e)

Make sure when you put your API key you do not include quotation marks

**Reply by sirlaurie** [reply]
there's no quotation mark. It works for openai, not for plugins

**Reply by danny-avila** [reply]
Are you submitting a key from the app or using the key from your .env file? It works in both cases for me.

Try restarting the server and refreshing the screen, followed by submitting the API key again. Also try it without any plugins selected

![image](https://github.com/danny-avila/LibreChat/assets/110412045/0c0cb6c5-0294-4c51-a575-cd653a8bc2f8)

**Reply by sirlaurie** [reply]
I did that. fresh update with shell script from document. submit key from app, without any plugin. log out and log in again.

but same error. 

how could I debug with this?

**Reply by danny-avila** [reply]
Check the debug and error logs located in `/api/logs`. they should be the most recently updated. Please share them. If you are using docker, you need to inspect the files in the container, and they should be in `app/api/logs`

**Reply by danny-avila** [reply]
Share the point of failure, especially important is the error log since it has the call stack trace.

**Comment by sirlaurie** [comment]
error log (error-2024-01-26.log)

```bash
{"level":"error","message":"[handleAbortError] AI response error; aborting request: Unexpected token 's', \"sk-NY9Cgvb\"... is not valid JSON","stack":"SyntaxError: Unexpected token 's', \"sk-NY9Cgvb\"... is not valid JSON\n    at JSON.parse (<anonymous>)\n    at initializeClient (/home/lori/LibreChat/api/server/services/Endpoints/gptPlugins/initializeClient.js:64:49)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async /home/lori/LibreChat/api/server/routes/ask/gptPlugins.js:160:24"}
```

**Reply by danny-avila** [reply, marked as answer]
I believe you might have `PLUGINS_USE_AZURE` enabled as you are hitting a condition only when that variable is enabled, or the word "azure" is detected in your API key. Is either of this the case? I would never expect the latter but not impossible

**Reply by sirlaurie** [reply]
indeed. I comment `PLUGINS_USE_AZURE`, now it works!

**Reply by danny-avila** [reply]
Great! I'm going to add some better handling here. It should be showing the Set Azure keys when it's enabled but it currently isn't: 
![image](https://github.com/danny-avila/LibreChat/assets/110412045/86d38550-9449-457e-b0e0-9cec2be5f0f6)

**Reply by danny-avila** [reply]
merging fix: https://github.com/danny-avila/LibreChat/pull/1649
